tests/parametric/test_otel_span_methods.py

```diff
@@ -348,3 +321,720 @@ def test_otel_get_span_context(self, test_agent, test_library):
+            with test_library.otel_start_span("otel_span_name", span_kind=SK_CONSUMER) as span:
+                span.set_attributes({"messaging.system": "kafka"})
+                span.set_attributes({"messaging.operation": "receive"})
+                span.end_span()
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, otel_span(name="otel_span_name"))
+        assert len(trace) == 1
+
+        root_span = get_span(test_agent)
+
+        assert root_span["name"] == "kafka.receive"
+        assert root_span["resource"] == "otel_span_name"
+
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "python_http", reason="Not implemented")
+    def test_otel_span_operation_name_message_producer(self, test_agent, test_library):
+        """
+            Tests that the operation name will be set to `messaging.system + "." + messaging.operation` when:
+            - Span kind is set to Producer
+            - messaging.system is set to something
+            - messaging.operation is set to something
+
+            (https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/messaging/)
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_span_name", span_kind=SK_PRODUCER) as span:
+                span.set_attributes({"messaging.system": "kafka"})
+                span.set_attributes({"messaging.operation": "publish"})
+                span.end_span()
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, otel_span(name="otel_span_name"))
+        assert len(trace) == 1
+
+        root_span = get_span(test_agent)
+
+        assert root_span["name"] == "kafka.publish"
+        assert root_span["resource"] == "otel_span_name"
+
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "python_http", reason="Not implemented")
+    def test_otel_span_operation_name_message_server(self, test_agent, test_library):
+        """
+            Tests that the operation name will be set to `messaging.system + "." + messaging.operation` when:
+            - Span kind is set to Server
+            - messaging.system is set to something
+            - messaging.operation is set to something
```

Comment:
I can't find this rule (with server kind) in the spec.

Reply:
Yep, there was a comment from @zacharycmontoya  a while ago about whether or not we should add the case for `server` kind for messaging, but I think it has gotten lost with all of my commits/reorganizing in this file

https://github.com/DataDog/system-tests/pull/1655#discussion_r1376729962

I've asked about whether or not it should actually be added as it wasn't in our spec but does appear to be a valid span kind in OpenTelemetry for messaging https://opentelemetry.io/docs/specs/semconv/messaging/messaging-spans/#span-kind

Reply:
Sure sounds good, we can add `SERVER` to the list. I'll update the doc.